include/galsim/SBInclinedExponentialImpl.h

```diff
@@ -0,0 +1,218 @@
+
+        /// @brief The fractional flux relative to the untruncated profile.
+        double getFluxFraction() const;
+
+        /**
+         * @brief The factor by which to multiply the returned value from xValue.
+         *
+         * Since the returned value needs to be multiplied by flux/r0^2 anyway, we also let
+         * the caller of xValue multiply by the normalization, which we calculate for them here.
+         */
+        double getXNorm() const;
+
+        /**
+         * @brief Shoot photons through unit-size, unnormalized profile
+         * Inclined profiles are sampled with a numerical method, using class
+         * `UniformDeviate`.
+         *
+         * @param[in] N  Total number of photons to produce.
+         * @param[in] ud UniformDeviate that will be used to draw photons from distribution.
+         * @returns PhotonArray containing all the photons' info.
+         */
+        boost::shared_ptr<PhotonArray> shoot(int N, UniformDeviate ud) const;
+
+    private:
+
+        InclinedExponentialInfo(const InclinedExponentialInfo& rhs); ///< Hide the copy constructor.
+        void operator=(const InclinedExponentialInfo& rhs); ///<Hide assignment operator.
+
+        // Input variables:
+        double _h_tani_over_r;
+        double _half_pi_h_tani_over_r;
+        const GSParamsPtr _gsparams; ///< The GSParams object.
+
+        // Some derived values calculated in the constructor:
+        double _ksq_max;   ///< If ksq < _kq_min, then use faster taylor approximation for kvalue
+        double _ksq_min;   ///< If ksq > _kq_max, then use kvalue = 0
+
+        // Parameters calculated when they are first needed, and then stored:
+        mutable double _maxk;    ///< Value of k beyond which aliasing can be neglected.
+        mutable double _stepk;   ///< Sampling in k space necessary to avoid folding.
+
+        // Parameters for the inverse Fourier transform
+        /* NYI
+        mutable Table2D<double,double> _ift;  ///< Lookup table for inverse Fourier transform.
+        */
+
+        // Classes used for photon shooting
+        /* NYI
+        mutable boost::shared_ptr<FluxDensity> _radial;
+        mutable boost::shared_ptr<OneDimensionalDeviate> _sampler;
+        */
+
+        // Helper functions used internally:
+        /* NYI
+        void buildIFT() const;
+        */
+    };
+
+    class SBInclinedExponential::SBInclinedExponentialImpl : public SBProfileImpl
+    {
+    public:
+        SBInclinedExponentialImpl(double i, double scale_radius, double scale_height, double flux,
+                 const GSParamsPtr& gsparams);
+
+        ~SBInclinedExponentialImpl() {}
+
+        double xValue(const Position<double>& p) const;
+        std::complex<double> kValue(const Position<double>& k) const;
+
+        double maxK() const;
+        double stepK() const;
+
+        void getXRange(double& xmin, double& xmax, std::vector<double>& splits) const
+        {
+            splits.push_back(0.);
```

Comment:
0 isn't much of a special point here.  Should remove this line.  (And in getYRange).